react/__tests__/Telemarketing.test.tsx

```diff
@@ -43,7 +43,128 @@ describe('<Telemarketing /> component', () => {
+        emailInput="email@vtex.com"
+        intl={intl}
+        loading={false}
+        onDepersonify={() => {}}
+        onInputChange={() => {}}
+        onSetSession={() => {}}
+      />
+    )
+
+    const attendantEmail = getByText(email)
+    expect(attendantEmail).toBeTruthy()
+  })
+
+  it('should show login button and form', () => {
+    const { getByText, getByPlaceholderText } = render(
+      <Telemarketing
+        attendantEmail="attendant@vtex.com"
+        emailInput="email@vtex.com"
+        intl={intl}
+        loading={false}
+        onDepersonify={() => {}}
+        onInputChange={() => {}}
+        onSetSession={() => {}}
+      />
+    )
+
+    const loginAs = getByText('Login as')
+    expect(loginAs).toBeTruthy()
+
+    const emailPlaceholder = getByPlaceholderText('Ex: example@mail.com')
+    expect(emailPlaceholder).toBeTruthy()
+
+    const login = getByText('Login')
+    expect(login).toBeTruthy()
+  })
+
+  it('should show client data', () => {
+    const client: Client = {
+      document: 'my client document',
+      phone: '+9999999999999',
+      name: 'my client name',
+      email: 'client@vtex.com',
+    }
+
+    const { getByText } = render(
+      <Telemarketing
+        attendantEmail="attendant@vtex.com"
+        client={client}
+        emailInput="email@vtex.com"
+        intl={intl}
+        loading={false}
+        onDepersonify={() => {}}
+        onInputChange={() => {}}
+        onSetSession={() => {}}
+      />
+    )
+
+    const document = getByText(client.document)
+    expect(document).toBeTruthy()
+
+    const phone = getByText(client.phone)
+    expect(phone).toBeTruthy()
+
+    const name = getByText(client.name)
+    expect(name).toBeTruthy()
+
+    const email = getByText(client.email)
+    expect(email).toBeTruthy()
+  })
+
+  it('should login when clicked', () => {
+    const onSetSession = jest.fn()
+    const { getByText } = render(
+      <Telemarketing
+        attendantEmail="attendant@vtex.com"
+        emailInput="email@vtex.com"
+        intl={intl}
+        loading={false}
+        onDepersonify={() => {}}
+        onInputChange={() => {}}
+        onSetSession={onSetSession}
+      />
+    )
+
+    fireEvent.click(getByText('Login'))
+
+    expect(onSetSession).toBeCalledTimes(1)
+  })
+
+  it('should logout when clicked', () => {
+    const onDepersonify = jest.fn()
+    const client: Client = {
+      document: 'my client document',
+      phone: '+9999999999999',
+      name: 'my client name',
+      email: 'client@vtex.com',
+    }
+
+    const { getByText } = render(
+      <Telemarketing
+        attendantEmail="attendant@vtex.com"
+        client={client}
+        emailInput="email@vtex.com"
+        intl={intl}
+        loading={false}
+        onDepersonify={onDepersonify}
+        onInputChange={() => {}}
+        onSetSession={() => {}}
+      />
+    )
+
+    fireEvent.click(getByText('Logout'))
+
+    expect(onDepersonify).toBeCalledTimes(1)
```

Comment:
You can use `toHaveBeenCalled()`

Reply:
I prefered to use `toBeCalledTimes` to garantee that one click only runs the function once.